token-metadata/program/src/processor/delegate/delegate.rs

```diff
@@ -405,6 +405,26 @@ fn create_persistent_delegate_v1(
         ],
     )?;
 
+    // For Utility Delegates we request Close Authority as well so that the
+    // token can be closed by the delegate on Burn.
+    if matches!(role, TokenDelegateRole::Utility) {
```

Comment:
> Does it let a Utility delegate do anything strange outside our burn handler if calling spl-token directly? 

I don't think so. SPL Token won't let you close the account if it has a token balance and pNFTs are frozen so the burn can only happen through our handler. After an Owner Transfer, the token balance for the account will be 0 but the Utility Delegate will be cleared, so only the owner can close it to clean it up.

> Would it be possible/desirable to instead request this spl authority inside of the burn handler, so that it is for sure only usable during the burn handler?

This is great question, and the reason it needs to happen in the `Delegate` instruction is that the owner has to sign to delegate close authority to another address and once we get to a `Burn` instruction where the authority is a Utility Delegate, we don't have an owner signer.

Reply:
Ok this all makes sense, thank you for the explanation.